pygmt/src/tilemap.py

```diff
@@ -0,0 +1,155 @@
+        Optional. The tile source: web tile provider or path to a local file.
+        Provide either:
+
+        - A web tile provider in the form of a
+          :class:`xyzservices.TileProvider` object. See
+          :doc:`Contextily providers <contextily:providers_deepdive>` for a
+          list of tile providers [Default is
+          ``xyzservices.providers.Stamen.Terrain``, i.e. Stamen Terrain web
+          tiles].
+        - A web tile provider in the form of a URL. The placeholders for the
+          XYZ in the URL need to be {{x}}, {{y}}, {{z}}, respectively. E.g.
+          ``https://{{s}}.tile.openstreetmap.org/{{z}}/{{x}}/{{y}}.png``.
+        - A local file path. The file is read with
+          :doc:`rasterio <rasterio:index>` and all bands are loaded into the
+          basemap. See
+          :doc:`contextily:working_with_local_files`.
+
+        IMPORTANT: Tiles are assumed to be in the Spherical Mercator projection
+        (EPSG:3857).
+
+    lonlat : bool
+        Optional. If ``False``, coordinates in ``region`` are assumed to be
+        Spherical Mercator as opposed to longitude/latitude [Default is
+        ``True``].
+
+    wait : int
+        Optional. If the tile API is rate-limited, the number of seconds to
+        wait between a failed request and the next try [Default is ``0``].
+
+    max_retries : int
+        Optional. Total number of rejected requests allowed before contextily
+        will stop trying to fetch more tiles from a rate-limited API [Default
+        is ``2``].
+
+    kwargs : dict
+        Extra keyword arguments to pass to :meth:`pygmt.Figure.grdimage`.
+
+    Raises
+    ------
+    ImportError
+        If ``rioxarray`` is not installed. Follow
+        :doc:`install instructions for rioxarray <rioxarray:installation>`,
+        (e.g. via ``pip install rioxarray``) before using this function.
+    """
+    kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
+
+    if rioxarray is None:
+        raise ImportError(
+            "Package `rioxarray` is required to be installed to use this function. "
+            "Please use `pip install rioxarray` or "
+            "`mamba install -c conda-forge rioxarray` "
+            "to install the package."
+        )
+
+    raster = load_tile_map(
+        region=region,
+        zoom=zoom,
+        source=source,
+        lonlat=lonlat,
+        wait=wait,
+        max_retries=max_retries,
+    )
+
+    # Reproject raster from Web Mercator (EPSG:3857) to lonlat (OGC:CRS84) if
+    # bounding box region was provided in lonlat
+    if lonlat and raster.rio.crs == "EPSG:3857":
+        raster = raster.rio.reproject(dst_crs="OGC:CRS84")
+        raster.gmt.gtype = 1  # set to geographic type
+
+    # Only set region if no_clip is False, so that plot is clipped to exact
+    # bounding box region
+    if not kwargs.get("N"):
+        kwargs["R"] = "/".join(str(coordinate) for coordinate in region)
```

Comment:
When `no_clip` is not given, it defaults to `None` and `not kwargs.get("N")` is True, which means the plot is clipped. I find the code is a little difficult to understand. Do you think the following changes make it better?
```suggestion
    # Only set region if no_clip is False or None, so that plot is clipped to exact
    # bounding box region
    if kwargs.get("N") is not True:
        kwargs["R"] = "/".join(str(coordinate) for coordinate in region)
```

Reply:
Yeah, double negatives are confusing. I went with `kwargs.get("N") in [None, False]` at 125574ee52f91624cb406769ee8b45297ebdb5f5.